File: sdk/node/src/api/course.module/course.module.model.ts

import { IsString, IsNumber, IsOptional, IsBoolean, IsUUID } from 'class-validator';

export class CourseModuleCreateModel {
    @IsOptional()
    @IsUUID()
    id?: string;  
    @IsString()
    Name: string;
    @IsString()
    Description: string;
    @IsString()
    ImageUrl: string;
    @IsNumber()
    DurationInMins: number;
    @IsNumber()
    Sequence: number;

}

export class CourseModuleUpdateModel {
    @IsOptional()
    @IsString()
    Name?: string;
    @IsOptional()
    @IsString()
    Description?: string;
    @IsOptional()
    @IsString()
    ImageUrl?: string;
    @IsOptional()
    @IsNumber()
    DurationInMins?: number;
    @IsOptional()
    @IsNumber()
    Sequence?: number;

}

export class CourseModuleSearchParams {
    @IsString()
    name: string;
}

export class ApiResponse<T> {
    Data: T;

    @IsOptional()
    @IsString()
    Message?: string;

    @IsOptional()
    @IsBoolean()
    Success?: boolean;
}

File: sdk/dotnet/lmsservice_db/src/api/course.module/CourseModuleModel.cs

using System.ComponentModel.DataAnnotations;

namespace lmsservice_db.src.api.course.module.course.modulemodel;
public class CourseModuleCreateModel
{
        [Required]
    [StringLength(64, MinimumLength = 0)]
    public string Name { get; set; } = string.Empty;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public string Description { get; set; } = string.Empty;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public string ImageUrl { get; set; } = string.Empty;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public int DurationInMins { get; set; } = 0;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public int Sequence { get; set; } = 0;


}

public class CourseModuleUpdateModel
{
        [Required]
    [StringLength(64, MinimumLength = 0)]
    public string Name { get; set; } = string.Empty;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public string Description { get; set; } = string.Empty;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public string ImageUrl { get; set; } = string.Empty;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public int DurationInMins { get; set; } = 0;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public int Sequence { get; set; } = 0;


}

public class CourseModuleSearchFilters
{
        [StringLength(64, MinimumLength = 0)]
    public string Name { get; set; } = string.Empty;

    [StringLength(64, MinimumLength = 0)]
    public string Description { get; set; } = string.Empty;

    [StringLength(64, MinimumLength = 0)]
    public string ImageUrl { get; set; } = string.Empty;

    [StringLength(64, MinimumLength = 0)]
    public int DurationInMins { get; set; } = 0;

    [StringLength(64, MinimumLength = 0)]
    public int Sequence { get; set; } = 0;


}

public class ApiResponse
{
    [Required]
    public Dictionary<string, object>? Data { get; set; }

    [StringLength(500, MinimumLength = 2)]
    public string Message { get; set; } = string.Empty;

    public bool? Success { get; set; }
}

